go-controller/pkg/metrics/ovn.go

```diff
@@ -279,6 +285,142 @@ func ovnControllerConfigurationMetricsUpdater() {
+			}
+			return 0
+		}))
+}
+
+func ovnControllerClusterSubnetMetricsPublisher() {
+	var cidrs []*net.IPNet
+
+	for _, subnet := range config.Default.ClusterSubnets {
+		cidrs = append(cidrs, subnet.CIDR)
+	}
+	cidrV4, cidrV6, v4Size, v6Size := getSubnetDetails(cidrs)
+
+	var metricClusterSubnet = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "cluster_subnets",
+		Help: "A metric with a constant '1' value labeled by subnets that " +
+			"specifies the subnets used for cluster. subnets_v4 and subnets_v6 are " +
+			"used as labels for IPv4 and IPv6 subnets respectively. If any of the subnets " +
+			"is not configured, corresponding label will have a blank value. If there are multiple " +
+			"subnets configured per IP family, then label value will be comma separated.",
+	},
+		[]string{
+			"subnets_v4",
+			"subnets_v6",
+		},
+	)
+	ovnRegistry.MustRegister(metricClusterSubnet)
+	metricClusterSubnet.WithLabelValues(cidrV4, cidrV6).Set(1)
+
+	var metricClusterV4Prefix = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "cluster_subnets_size_ipv4",
+		Help: "A metric with a constant '1' value labeled by prefix-length" +
+			"Value of the lable is the size of IPv4 cluster subnets. If there are multiple IPv4 " +
+			"subnets configured then lable value will be comma separated. Empty value for the label " +
+			"means IPv4 subnet is not configured.",
+	},
+		[]string{
+			"prefix-length",
+		},
+	)
+	ovnRegistry.MustRegister(metricClusterV4Prefix)
+	metricClusterSubnet.WithLabelValues(v4Size).Set(1)
+
+	var metricClusterV6Prefix = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "cluster_subnets_size_ipv6",
+		Help: "A metric with a constant '1' value labeled by prefix-length" +
+			"Value of the lable is the size of IPv6 cluster subnets. If there are multiple IPv6 " +
+			"subnets configured then lable value will be comma separated. Empty value for the label " +
+			"means IPv6 subnet is not configured.",
+	},
+		[]string{
+			"prefix-length",
+		},
+	)
+	ovnRegistry.MustRegister(metricClusterV6Prefix)
+	metricClusterSubnet.WithLabelValues(v6Size).Set(1)
+}
+
+func getSubnetDetails(cidrs []*net.IPNet) (string, string, string, string) {
+	var cidrV4 = ""
+	var cidrV6 = ""
+	var v4Size = ""
+	var v6Size = ""
+
+	for _, cidr := range cidrs {
+		ones, _ := cidr.Mask.Size()
+		if utilnet.IsIPv6CIDR(cidr) {
+			if cidrV6 == "" {
+				cidrV6 = cidr.String()
+				v6Size = string(ones)
+			} else {
+				cidrV6 = cidrV6 + "," + cidr.String()
+				v6Size = v6Size + "," + string(ones)
+			}
+		} else {
+			if cidrV4 == "" {
+				cidrV4 = cidr.String()
+				v4Size = string(ones)
+			} else {
+				cidrV4 = cidrV4 + "," + cidr.String()
+				v4Size = v4Size + "," + string(ones)
```

Comment:
The code is broken, on L315 and L329 you do `strconv.Atoi(v6Size)` which will break because of the comma.  We might need one metric detailing how many cluster subnets have been defined and other metrics for the size of each.

Reply:
I dont think so. For service subnets we will allow only one v4 and v6 subnets. I am doing Atoi only for service subnets which will only have one value or blank. In case of blank we will default to zero.  So this code works for service subnets which will have at max one subnet for v4 and v6 for which we expose a numeric metric as well as cluster subnets for which metric is exposed as string via label.